tldw_Server_API/app/api/v1/endpoints/character_chat_sessions.py

```
@@ -6118,50 +6281,108 @@ async def persist_streamed_assistant_message(
+                parent_message_id=body.user_message_id,
+                speaker_name=resolved_speaker_name,
+            )
+        elif persist_fingerprint is not None:
+            existing_persist = _find_existing_stream_persist_message(db, chat_id, persist_fingerprint)
+        if existing_persist is not None:
+            existing_id, existing_extra = existing_persist
+            if existing_extra.get("persist_validation_degraded"):
+                raise HTTPException(
+                    status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+                    detail=_build_persist_validation_degraded_detail(existing_id),
+                )
```

Ruff (0.15.9)

[warning] Abstract `raise` to an inner function

(TRY301)